src/enable_variable_ops/ngraph_enter_in_catalog.cc

```diff
@@ -0,0 +1,145 @@
+/*******************************************************************************
+ * Copyright 2019-2020 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#pragma once
+
+#include "tensorflow/core/graph/algorithm.h"
+#include "tensorflow/core/graph/graph.h"
+
+#include "ngraph/ngraph.hpp"
+#include "ngraph/serializer.hpp"
+#include "ngraph_catalog.h"
+#include "ngraph_log.h"
+#include "ngraph_utils.h"
+
+using namespace std;
+namespace ng = ngraph;
+
+namespace tensorflow {
+
+namespace ngraph_bridge {
+
+// Used by only NGraphVar or NGraphAssign
+Status GetSharedName(Node* node, string* shared_name) {
+  if (node->type_string() == "NGraphVariable") {
+    TF_RETURN_IF_ERROR(GetNodeAttr(node->attrs(), "shared_name", shared_name));
+    if (shared_name->empty()) {
+      (*shared_name) = node->name();
+    }
+    return Status::OK();
+  }
+
+  auto temp = node;
+  while (temp->type_string() != "NGraphVariable") {
```

Comment:
Maybe we can add a comment here, explaining the need for the while loop. That we are backtracking through "modifier" nodes like Assigns, till we hit the underlying "modifiable" node NgVariable

Also maybe a comment that eventually this loop is guaranteed to terminate because the backtracking `temp->input_node(0, &input_0))` will run out in a finite graph at some point